### ISSUE TITLE
Document branching strategy

### ISSUE THREAD

**Issue by flanakin** [microsoft/cloud-hubs#65]
## ⚠️ Problem
Everyone is currently working out of main but that will eventually cause issues when features have interim PRs that are not ready to roll out yet.

## 🛠️ Solution
Document a branching strategy where devs work out of their own branches/forks, then commit back to a feature branch, which then gets merged into a **dev** branch. When the release is ready, we'll merge **dev** to **main**.

## ℹ️ Additional context
This should be done before closing any PRs that will not be included in the first release.

## 🙋‍♀️ Ask for the community

We could use your help:
1. Please vote this issue up (👍) to prioritize it.
2. Leave comments if you'd like to share any thoughts to improve the plans.
